linux/keyman-config/keyman_config/uninstall_kmp.py

```diff
@@ -82,14 +82,20 @@ def _uninstall_kmp_common(location, packageID, removeLanguages):
 
 def _uninstall_keyboards_from_ibus(keyboards, packageDir):
     bus = get_ibus_bus()
-    if bus or os.environ.get('SUDO_USER'):
-        # install all kmx for first lang not just packageID
-        for kb in keyboards:
-            ibus_keyboard_id = get_ibus_keyboard_id(kb, packageDir)
-            uninstall_from_ibus(bus, ibus_keyboard_id)
-        restart_ibus(bus)
-    else:
-        logging.warning("could not uninstall keyboards from IBus")
+    ibusUtil = IbusUtil()
+    sources = ibusUtil.read_preload_engines()
+    if not sources:
+        return
+
+    # install all kmx for first lang not just packageID
```

Comment:
```suggestion
    # uninstall each specified kmx for all languages
```

Or something similar.

Reply:
Done